utilities/copyoffload_migration.py

```
@@ -1493,21 +1522,92 @@ def _verify_throttled_events_on_pods(
+
+    Raises:
+        ValueError: If populate pod count does not exceed the limit, or too few PVCs have
+            PopulatorThrottled events.
+    """
+    pod_count = len(pod_logs)
+    min_expected_throttled = pod_count - max_populator_inflight
+    if min_expected_throttled <= 0:
+        raise ValueError(
+            f"Expected more populate pods ({pod_count}) than in-flight limit "
+            f"({max_populator_inflight}) to verify throttling for migration '{migration_uid}'"
+        )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)